tool/tsh/tsh.go

```diff
@@ -4530,6 +4538,20 @@ func setSiteNameFromEnv(cf *CLIConf, fn envGetter) {
 	}
 }
 
+// setUsernameFromEnv sets teleport user name from environment if configured.
+func setUsernameFromEnv(cf *CLIConf, fn envGetter) {
+	if userName := fn(teleport.SSHTeleportUser); userName != "" {
+		cf.Username = userName
+	}
+}
+
+// setUserFromEnv sets teleport user name from environment if configured.
+func setProxyFromEnv(cf *CLIConf, fn envGetter) {
+	if proxy := fn(teleport.SSHSessionWebproxyAddr); proxy != "" {
+		cf.Proxy = proxy
+	}
+}
+
```

Comment:
Do we even need these functions (plus the other ones like `setSiteNameFromEnv` and `setKubernetesClusterFromEnv`?

We only call them if the field we're trying to set is not empty and then all they do is check if the value from the environment is not empty and then set it. But even if it was empty, it wouldn't change the outcome.

Seems like we can simplify `setEnvFlags` to be something like the following. What do you think?

```go
func setEnvFlags(cf *CLIConf, getEnv envGetter) {
  // [...]
	
  if cf.Username == "" {
    cf.Username = getEnv(teleport.SSHTeleportUser)
  }

  // [...]
}
```

Reply:
Also, extremely minor but I'd rename `fn` to `getEnv` so it's easier for the reader to understand what the function being called is going to do (read value from the environment).